# Note: You MUST load these PID parameters for all joints that are using the
# PositionJointInterface, otherwise the arm + gripper will act like a giant
# parachute, counteracting gravity, and causing some of the wheels to lose
# contact with the ground, so the robot won't be able to properly navigate. See
# https://github.com/ros-simulation/gazebo_ros_pkgs/issues/602
gazebo_ros_control:
  pid_gains:
    # these gains are used by the gazebo_ros_control plugin
 
# Left Arm 
    left_arm_shoulder_roll_joint:       &arm_1_gains {p: 500, d: -0.01, i: 0.1,   i_clamp: 3, torque_clamp: 60}      
    left_arm_shoulder_pitch_joint:      &arm_2_gains {p: 500, d: -0.01, i: 0.1,   i_clamp: 3, torque_clamp: 50}           
    left_arm_shoulder_yaw_joint:        &arm_3_gains {p: 500, d: -0.01, i: 0.1,   i_clamp: 3, torque_clamp: 50}      
    left_arm_elbow_pitch_joint:         &arm_4_gains {p: 500, d: -0.01, i: 0.1,   i_clamp: 3, torque_clamp: 50}      
    left_arm_elbow_yaw_joint:           &arm_5_gains {p: 500, d: -0.01, i: 0.1,   i_clamp: 3, torque_clamp: 50}
    left_arm_wrist_pitch_joint:         &arm_6_gains {p: 2,   d: 0.01, i: 0.01,   i_clamp: 3, torque_clamp: 50}     
    left_arm_wrist_roll_joint:          &arm_7_gains {p: 80,  d: -0.01, i: 0.1,   i_clamp: 3, torque_clamp: 50}
      
      
# Right Arm      
    right_arm_shoulder_roll_joint:      *arm_1_gains      
    right_arm_shoulder_pitch_joint:     *arm_2_gains     
    right_arm_shoulder_yaw_joint:       *arm_3_gains      
    right_arm_elbow_pitch_joint:        *arm_4_gains      
    right_arm_elbow_yaw_joint:          *arm_5_gains     
    right_arm_wrist_pitch_joint:        *arm_6_gains      
    right_arm_wrist_roll_joint:         *arm_1_gains     
 
# # Grippers 
    left_arm_gripper_joint:             &gripper_1_gains {p: 10000, d:  5.0, i:  1.0, i_clamp: 3}      
    left_arm_gripper_joint2:            *gripper_1_gains    
    right_arm_gripper_joint:            *gripper_1_gains #&gripper_2_gains {p: 10000, d: -0.01, i: 0.1, i_clamp: 3}      
    right_arm_gripper_joint2:           *gripper_1_gains     
      
# # PTU    
    ptu_pan_joint:                      &ptu_gains {p: 100, d: 1, i: 0.1, i_clamp: 3, torque_clamp: 50}      
    ptu_tilt_joint:                     *ptu_gains
      

# # Wheels      
      
    wheel_right_joint:                  &wheel_gains {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0}      
    wheel_left_joint:                   *wheel_gains
